### DISCUSSION TITLE
Error response status codes for webhook triggered flows

### DISCUSSION THREAD

**Discussion by rishikeshjoshi** [directus/directus#19178] [Feature Requests]
### Summary

As discussed in the request https://github.com/directus/directus/discussions/16342, you can throw an error or ensure that the flow ends on the rejection path to return a custom error status code. However this does not work with flows triggered via webhooks and it would be good to introduce a feature to support that. 

### Motivation

We enhance the flows support to allow webhook triggered flows to send custom response status codes and payload.

### Detailed Design

In the flow manager class 

```typescript
// api/src/flows.ts

// Works only in case of event trigger filters. Add support for webhook triggered flows to throw custom errors.
if (flow.trigger === 'event' && flow.options['type'] === 'filter' && lastOperationStatus === 'reject') {
  throw keyedData[LAST_KEY];
}

if (flow.options['return'] === '$all') {
  return keyedData;
} else if (flow.options['return']) {
  return get(keyedData, flow.options['return']);
}

return undefined;
``` 

### Must Have:
- Implement support for webhook flows to throw errors.

### Should Have:
- Visibility of the error type.

### Drawbacks

- The structure of thrown error is not visible. There is no visibility about how this error(s) is/are to be translated to the caller (e.g http status code, body).

### Alternatives

Currently the user is not able to have custom logic and send an error status code to the caller.

### Adoption Strategy

Directus developers will need to update their flow consumers according to the response they send from their configured flows.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for the feature request. While this didn't reach the 15 upvotes in 90 days requirements, there's actually another feature request that requires this to be implemented in order to work which did, so lets consolidate the two and keep the discussion going over on https://github.com/directus/directus/discussions/18634
